test/infrastructure/docker/api/v1beta1/dockercluster_types.go

```diff
@@ -88,10 +88,13 @@ type DockerClusterStatus struct {
 // APIEndpoint represents a reachable Kubernetes API endpoint.
```

Comment:
(not related to this file, just had to put the comment somewhere)

I think it would be good to have e2e test coverage for this feature.

WDYT about setting a non-default port in `test/e2e/data/infrastructure-docker/v1beta1/main/bases/cluster-with-kcp.yaml`?

I think this should make it easier to keep this working over time.

I assume when we don't make the Host/Port fields optional we probably have to set the host to an empty string (?)
(not sure if it works or it's rejected by the OpenAPI validation)